client/Reviews.jsx

```diff
@@ -0,0 +1,235 @@
+      .then(response => {
+        console.log('this is data ', response.data)
+        this.setState({
+          data: response.data,
+          copyData: response.data,
+          count: response.data.length,
+          pageReviews: response.data.slice(0, 7)
+        })
+      })
+      .catch(err => { // did not handle possible errors before
+        console.error("error in setting state")
+      });
+  };
+
+  allAverage(array) {
+    let count = 0;
+    let divide = array.length * 6;
+
+    array.map(obj => {
+      for (var key in obj.ratings) {
+        count += obj.ratings[key];
+      }
+    })
+
+    return count / divide;
+  };
+
+  nextPageClick() {
+    let list = this.state.pageReviews;
+    let length = list.length - 1;
+    let lastReview = list[length];
+    let lastReviewNum = lastReview.reviewNum;
+    let start;
+    let result = [];
+    if (list.length === 7) { // prevents nextPageClick from rendering a blank page if at the end of total list of reviews
+      for (var i = 0; i < this.state.data.length; i++) {
+        var temp = this.state.data[i];
+        if (temp.reviewNum === lastReviewNum) {
+          if ((i + 7) < (this.state.data.length - 1)) {
+            start = i + 1;
+            let end = start + 7;
+
+            while (start < end) {
+              result.push(this.state.data[start])
+              start++;
+            };
+
+            this.setState({
+              pageReviews: result
+            })
+            break;
+          } else {
+            for (var j = i + 1; j < this.state.data.length; j++) {
+              result.push(this.state.data[j])
+            }
+            this.setState({
+              pageReviews: result
+            })
+            break;
+          }
+        }
+      };
+    }
+
+
+
+  };
+
+  previousPageClick() {
+    let list = this.state.pageReviews;
+    let firstReview = list[0];
+    let firstReviewNum = firstReview.reviewNum;
+    let start;
+
+    for (var i = 0; i < this.state.data.length; i++) {
+      var temp = this.state.data[i];
+      if (temp.reviewNum === firstReviewNum) {
+        if (i === 0) {
+          start = 0;
+          break;
+        }
+        start = i - 1;
+      }
+    };
+
+    if (start !== 0) {
+      let end = start - 7;
+      let result = [];
+
+      while (start > end) {
+        result.unshift(this.state.data[start]);
+        start--;
+      };
+
+      this.setState({
+        pageReviews: result
+      })
+    }
+
+  };
+
+  handleChange(event) {
+    this.setState({
+      search: event.target.value
+    })
+  };
+
+  handleSubmit(event) {
+    let result = [];
+    let word = this.state.search;
+
+    for (var i = 0; i < this.state.data.length; i++) {
+      var temp = this.state.data[i];
+      if (temp.guestReview.includes(word)) {
+        result.push(temp);
+      }
+    }
+
+    console.log('this is data after search ', result)
+
+    if (result.length > 7) {
+      this.setState({
```

Comment:
this could just be style, but you could consider setting state outside of the if / else to prevent having repeated code. Then within the if / else handle the logic of pageReviews, since it seems like thats the only thing changing between the two conditional logic

Reply:
I see what you've pointed out here but in this instance setting state is different based on the if statement but I have noticed spots in my code where I could possibly remove the repeating setState call.